aiidalab_widgets_base/structures.py

```diff
@@ -702,33 +712,63 @@ def _rdkit_opt(self, smiles, steps):
 
         smiles = smiles.replace("[", "").replace("]", "")
         mol = Chem.MolFromSmiles(smiles)
+        if mol is None:
+            self.output.value = (
+                "WARNING: RDKit failed to parse SMILES, trying OpenBabel."
+            )
+            return None
         mol = Chem.AddHs(mol)
 
-        AllChem.EmbedMolecule(mol, maxAttempts=20, randomSeed=42)
+        params = AllChem.ETKDG()
+        params.randomSeed = 42
+        params.maxAttempts = 20
+        conf_id = AllChem.EmbedMolecule(mol, params=params)
+        if conf_id == -1:
+            # useRandomCoords might be more robust for larger molecules
+            # https://sourceforge.net/p/rdkit/mailman/message/21776083/
+            params.useRandomCoords = True
+            conf_id = AllChem.EmbedMolecule(mol, params=params)
+        if conf_id == -1:
+            msg = "WARNING: RDkit embedding failed, trying OpenBabel"
+            self.output.value = msg
+            return None
+
+        if not AllChem.UFFHasAllMoleculeParams(mol):
+            msg = "WARNING: RDKit is missing UFF parameters, trying OpenBabel"
+            self.output.value = msg
+            return None
+
         AllChem.UFFOptimizeMolecule(mol, maxIters=steps)
-        positions = mol.GetConformer().GetPositions()
+        positions = mol.GetConformer(id=conf_id).GetPositions()
         natoms = mol.GetNumAtoms()
         species = [mol.GetAtomWithIdx(j).GetSymbol() for j in range(natoms)]
         return self._make_ase(species, positions, smiles)
 
-    def _mol_from_smiles(self, smiles, steps=10000):
+    def _mol_from_smiles(self, smiles, steps=1000):
```

Comment:
10000 steps seems excessive. Probably better to bail out early rather than letting user wait forever.

Reply:
the correct geometry for: C12=CC=CC3=C1C([C]C4=C3C5=C(C(C(C6=CC=CC7=C6C([C]8)=CC=C7)=C8C=C9)=C9C=C%10)C%10=CC=C5C=C4)=CC=C2

is:

![image](https://user-images.githubusercontent.com/22955065/152409398-26d5428d-d860-4acd-bc76-d849ee66f9e2.png)

the one that fails would have two additional H atoms